### PR TITLE
Bump MSRV to 1.63.0

### LINKED ISSUE #959: Increase MSRV to 1.63
This will allow us to progress on https://github.com/rust-itertools/itertools/labels/const-generics, particularly #560. This entails:
- updating
  https://github.com/rust-itertools/itertools/blob/a0225ceee50cafe9e974699e1cbf7fa9a1dcf901/.github/workflows/ci.yml#L47
- updating 
  https://github.com/rust-itertools/itertools/blob/a0225ceee50cafe9e974699e1cbf7fa9a1dcf901/Cargo.toml#L19
- updating
  https://github.com/rust-itertools/itertools/blob/a0225ceee50cafe9e974699e1cbf7fa9a1dcf901/src/lib.rs#L49
- auditing for `allow`'d clippy lints that can now be fixed 
- auditing `TODO` comments that can now be fixed

### PR DESCRIPTION
**Pull request by kinto-b**
Resolves #959

### CODE REVIEW

File: src/concat_impl.rs [hunk 1]
```diff
@@ -1,8 +1,6 @@
-use crate::Itertools;
-
 /// Combine all an iterator's elements into one element by using [`Extend`].
 ///
-/// [`IntoIterator`]-enabled version of [`Itertools::concat`].
+/// [`IntoIterator`]-enabled version of [`crate::Itertools::concat`].
```
Comment (Philippe-Cholet) [inline src/concat_impl.rs:3]: Usually, we do ``[`Itertools::...`](crate::Itertools::...)``. It's longer but visually nicer (once rendered): no `crate::` seen.
Same twice in the "kmerge.." file.

File: src/lib.rs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#![warn(missing_docs, clippy::default_numeric_fallback)]
```
Comment (Philippe-Cholet) [inline src/lib.rs:1]: We will keep this one forever, it's a safeguard against errors. See #932 if curious.

File: tests/quick.rs [hunk 3]
```diff
@@ -253,7 +253,6 @@ where
         let mut it = get_it();
 
         for _ in 0..(counts.len() - 1) {
-            #[allow(clippy::manual_assert)]
```
Comment (Philippe-Cholet) [inline tests/quick.rs:256]: Ah this lint was in "style" category (so warn by default) but was moved in "pedantic" (allow by default), nice.

**Review by Philippe-Cholet — CHANGES_REQUESTED** [review]
That was fast, thanks! Not much to change.

File: src/lib.rs (hunk 2 above)
Comment (kinto-b) [reply src/lib.rs]: Ah sorry misread the warn as allow 😳

File: src/concat_impl.rs (hunk 1 above)
Comment (kinto-b) [reply src/concat_impl.rs]: Ah good point, thanks :)

**Comment by kinto-b** [comment]
Haha, I've got a bit of free time on my hands at the moment (moved country and between jobs) and wasn't up to much when I saw your @ :) 

Thanks as always for your help!

**Review by Philippe-Cholet — APPROVED** [review]

**Pull request merged**